### PR TITLE
feature: allow forced update through GUI

### PR DESCRIPTION
**Pull request by cmotelet**
- allow update through GUI even if server installed in non-standard path
- add missing environment variables in readme.md file (some remains to be commented).

### CODE REVIEW

**Comment by sbender9** [comment]
Will the update actually work if it's in a non-standard location? We just run `npm install -g ...`

**Comment by cmotelet** [comment]
@sbender9 
`non-standard path` may not be the correct words, I would have to say in an unlisted path.
My target is the installation and update under Windows.
For good or bad reasons we can set the path of the modules and thus of the server out of your fixed list.
https://www.davidyardy.com/blog/change-default-global-installation-directory-for-nodejs-on-windows/
Another more global alternative would be to search the server like this: `npm list -g --depth 0 signalk-server@1.23.0`
`1.23.0` in the example, because knows the version already installed.
And that way, we know if the server is installed `globally`.

**Comment by tkurki** [comment]
@cmotelet just to be clear: is it so that with this change you can set the server up so that it can update itself, even if it is installed somewhere where the user has chosen?

File: README.md [hunk 1]
```diff
@@ -114,22 +114,32 @@ In this case, the server would look for the settings file at `/usr/local/etc/nod
...
+- `TCPSTREAMADDRESS` override the address the Signal K Stream (deltas) over TCP is listening on.
+- `DISABLEPLUGINS` disable all plugins so that they can not be enabled.
+- `DEFAULTENABLEDPLUGINS` a comma separated list of plugin ids that are overridden to be enabled by default if no setttings exist. lower preference than `DISABLEPLUGINS`.
+- `PLUGINS_WITH_UPDATE_DISABLED` a comma separated list of plugin that will not be updated.
+- `SECURITYSTRATEGY` override the security strategy module name.
+- `WSCOMPRESSION` compress websocket messages.
 - `MAXSENDBUFFERSIZE` the maximum number of bytes allowed in the server's send buffer of a WebSocket connection. The connection will be terminated if this is exceeded. Guards against slow or dysfunctional clients that can not cope with the message volume. Default is 512 * 1024 bytes.
-
+- `SIGNALK_SERVER_IS_UPDATABLE` allows the server to be updated through the GUI even if it is not installed in the standard paths.
+- `SIGNALK_DISABLE_SERVER_UPDATES` invalidates the ability to update the server through the GUI.
```
Comment (tkurki) [inline README.md]: _disables server updates in the GUI_  would be shorter and clearer

File: README.md [hunk 2]
```diff
@@ -114,22 +114,32 @@ In this case, the server would look for the settings file at `/usr/local/etc/nod
...
+- `DEBUG` a comma-separated list of tags for debugging the specified module (For example: signalk-server*,signalk-provider-tcp). Can now be defined directly in the graphical interface.
```
Comment (tkurki) [inline README.md:135]: I think we could also link to the debug module in npm or in Github.

File: README.md [hunk 3]
```diff
@@ -114,22 +114,32 @@ In this case, the server would look for the settings file at `/usr/local/etc/nod
...
+- `IS_IN_DOCKER` Automatically enabled if the server is running under docker and disables updating process through the GUI.
```
Comment (tkurki) [inline README.md]: Not really automatically, maybe _used to tell the server it is in Docker and not normally updateable_

File: README.md [hunk 4]
```diff
@@ -114,22 +114,32 @@ In this case, the server would look for the settings file at `/usr/local/etc/nod
...
+- `NPMREGISTRYTIMEOUT` allow adjust the timeout to acces to NPN web site (by default 20s)
```
Comment (tkurki) [inline README.md]: typo

good idea to have the default here

no need to rewrite what the variable does, but why you would want to use it: _how long to wait for the registry when retrieving the App Store listing_

File: README.md [hunk 5]
```diff
@@ -114,22 +114,32 @@ In this case, the server would look for the settings file at `/usr/local/etc/nod
...
+- `SECRETKEY` remains to be commented...
+- `ALLOW_DEVICE_ACCESS_REQUESTS` remains to be commented...
```
Comment (tkurki) [inline README.md]: needs default value.  I think a link to https://signalk.org/specification/1.4.0/doc/access_requests.html would be enough documentation

File: README.md [hunk 6]
```diff
@@ -114,22 +114,32 @@ In this case, the server would look for the settings file at `/usr/local/etc/nod
...
+- `ALLOW_NEW_USER_REGISTRATION` remains to be commented...
+- `ADMINUSER` remains to be commented...
+- `PRESERIALCOMMAND` remains to be commented...
```
Comment (tkurki) [inline README.md]: _Command to run [before opening a serial port )](https://github.com/SignalK/signalk-server-node/pull/666)_

**Comment by cmotelet** [comment]
@tkurki ,
With this change you can update the server itself but the server must be installed in path that can be supported by  `npm install -g signalk-server` command.
And the path is not necessarily in the fixed list set in `src/interfaces/appstore.js`
```
const npmServerInstallLocations = [
  '/usr/bin/signalk-server',
  '/usr/lib/node_modules/signalk-server/bin/signalk-server',
  '/usr/local/bin/signalk-server',
  '/usr/local/lib/node_modules/signalk-server/bin/signalk-server'
]
```

**Pull request merged**